backend/src/routes/vector.ts

```diff
@@ -1,133 +1,59 @@
-import { FieldValue } from "firebase-admin/firestore";
-import { createId } from "@paralleldrive/cuid2";
-import conn from "../db/db";
-import { customers } from "../db/schema/customers";
-import { eq } from "drizzle-orm";
 import vectorRoute from "./vector.route";
 import getFirebaseApp from "../firebase";
 import env from "../env";
 import { OpenAPIHono } from "@hono/zod-openapi";
+import { faceAuthController } from "../controller/face-auth-controller";
+import { getFaceEmbedding } from "../infra/face-embedding-repo";
+import { authenticateFace, registerEmbedding } from "../infra/face-auth-repo";
+import { createDBCustomer, findDBCustomerById } from "../infra/customer-repo";
+import { toHTTPException } from "./shared/exception";
+import { authFace } from "../service/face-auth-service";
+import { registerCustomer } from "../service/customer-service";
+import { registerCustomerController } from "../controller/customer-controller";
+import { validateCustomer } from "../domain/customer";
 
 const app = new OpenAPIHono();
 
-type EmbeddingResponse = {
-  embedding: number[][];
-};
-
 app.openapi(vectorRoute.authenticateFace, async (c) => {
   const formData = await c.req.formData();
-  const image = formData.get("image") as File | null;
-  if (!image) {
-    throw new Error("No image provided");
-  }
+  const image = formData.get("image") as File;
```

Comment:
imo: `c.req.valid('form')` を使用すると型定義付きで image が取得できるかと思います！
```suggestion
  const { image } = c.req.valid('form');
```

Reply:
うーん、これやりたかったんだけど、エラー出ちゃう
```sh
Argument of type '"form"' is not assignable to parameter of type 'never'.ts(2345)
```

Reply:
`vector.route.ts` の `authenticateFace` を以下のように書き換えるといけましたー
```typescript
const authenticateFace = createDefaultRoute({
  method: "post",
  path: "/face-auth",
  tags,
  validateResponse: true,
  operationId: "authenticateFace",
  description: "Authenticate a user using face recognition",
  request: {
    body: {
      content: {
        "multipart/form-data": {
          schema: z.object({
            image: z.instanceof(File).describe("Image for face authentication"),
          }),
        },
      },
    },
  },
  // ...
```